src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -304,14 +296,13 @@ public void AddRange(T[] items, int length)
                 Requires.NotNull(items, "items");
                 Requires.Range(length >= 0, "length");
 
+                if (length > items.Length)
+                    throw new IndexOutOfRangeException("length");
```

Comment:
Personally I'd be OK with the change in behavior by throwing the more appropriate exception. Folks shouldn't be catching the old exception anyway.